### PR TITLE
Get latest gis dataset based on folder name

### PR DESCRIPTION
**Pull request by fvankrieken**
baby pr - we shouldn't rely on what's in staging folder ideally

This logic is currently only used in ingest

### CODE REVIEW

File: dcpy/connectors/edm/publishing.py [hunk 1]
```diff
@@ -380,21 +381,31 @@ def _assert_gis_dataset_exists(name: str, version: str):
...
-    ## by that logic, should likely try to parse str version to dcpy version object
-    ## so we don't have strange cases of an unexpected subfolder (that starts with an _)
-    ## this would let us not have to supply versions for geosupport datasets ("24a", etc)
-    metadata = s3.get_metadata(BUCKET, _gis_dataset_path(dataset_name, "staging"))
-    version = metadata.last_modified.strftime("%Y%m%d")
+    """
+    Get latest version of GIS-published dataset in edm-publishing/datasets
+    assuming versions are sortable
+    """
+    gis_version_formats = [r"^\d{2}[A-Z]$", r"^\d{8}$"]
```
Comment (alexrichey) [inline dcpy/connectors/edm/publishing.py:388]: Should we move this over to dcpy.utils.versions?

File: dcpy/test/connectors/edm/test_publishing.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ def add_gis_datasets(create_buckets):
     test_objects = [
         f"datasets/{TEST_GIS_DATASET}/staging/{TEST_GIS_DATASET}.zip",
         f"datasets/{TEST_GIS_DATASET}/{today}/{TEST_GIS_DATASET}.zip",
-        f"datasets/{TEST_GIS_DATASET}/{TEST_VERSION.upper()}/{TEST_GIS_DATASET}.zip",
+        f"datasets/{TEST_GIS_DATASET}/20240101/{TEST_GIS_DATASET}.zip",
```
Comment (alexrichey) [inline dcpy/test/connectors/edm/test_publishing.py:27]: nit: let's pull this out into a constant

**Review by alexrichey — APPROVED** [review]
Two optional nits, but looks good!

File: dcpy/connectors/edm/publishing.py (hunk 1 above)
Comment (fvankrieken) [reply dcpy/connectors/edm/publishing.py:388]: Maybe. We need to make that more consistent and really implement it across things, so I'd rather just leave for now

**Pull request merged**